libminifi/src/core/flow/StructuredConfiguration.cpp

```diff
@@ -620,20 +628,26 @@ void StructuredConfiguration::parseRPGPort(const Node& port_node, core::ProcessG
 }
 
 void StructuredConfiguration::parsePropertyValueSequence(const std::string& property_name, const Node& property_value_node, core::ConfigurableComponent& component) {
+  core::Property myProp(property_name, "", "");
+  component.getProperty(property_name, myProp);
+
   for (const auto& nodeVal : property_value_node) {
     if (nodeVal) {
       Node propertiesNode = nodeVal["value"];
-      // must insert the sequence in differently.
-      const auto rawValueString = propertiesNode.getString().value();
-      logger_->log_debug("Found {}={}", property_name, rawValueString);
+      auto rawValueString = propertiesNode.getString().value();
+      if (myProp.isSensitive()) {
+        rawValueString = decryptProperty(rawValueString);
+      }
+      logger_->log_debug("Found property {}", property_name);
```

Comment:
Could this be changed in the future to allow decrypting properties that are _not_ marked as sensitive, without breaking backwards compatibility?
I'm thinking about a scenario where we accidentally marked a property as sensitive that's not really sensitive, so we make the property non-sensitive, but existing configs have the encrypted value after upgrade to this hypothetical future version.

I'm not requesting the change now, just want to make sure that should we ever make a property non-sensitive, we can also make this change then.

Reply:
Yes, good point.  We could even make the change now; the one drawback is that no properties would be allowed to start with `enc{` and end with `}`.  I don't know if that is a big problem.  Alternatively, we could introduce a new (and possibly temporary) `wasSensitive()` flag if/when we change a property from sensitive to non-sensitive.

Reply:
I don't think it's a big problem to break properties that use the `enc{`/`}` wrappers on their own, but it's also not ideal. To limit the set of broken property values, we could check the format of the string inside the wrapper, and only act if it looks like what our encrypted values look.

Maybe it's best to leave it as is for now, and figure this out later if/when necessary.